app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/BusinessIdentificationController.scala

```diff
@@ -249,9 +237,14 @@ class BusinessIdentificationController @Inject()(
 
                     lookupNextPage(initialDetails)
                   }
-                case No =>
-                  Future.successful(routes.BusinessIdentificationController.submitBusinessDetailsForm(
-                    request.session.get("businessType")))
+                case No => {
+                  request.session
+                    .get("businessType")
+                    .map(typeFound =>
+                      Future.successful(routes.BusinessIdentificationController.submitBusinessDetailsForm(
+                        IdentifyBusinessType.apply(typeFound))))
+                    .getOrElse(Future.successful(routes.BusinessIdentificationController.showBusinessTypeForm()))
```

Comment:
Just noticed this, and I can see it was there before, but shouldn't this be redirecting to `showBusinessDetailsForm` rather than `submitBusinessDetailsForm`?